mesonbuild/backend/ninjabackend.py

```diff
@@ -3632,6 +3649,9 @@ def generate_link(self, target: build.BuildTarget, outname, obj_list, linker: T.
             elem.add_item('ARGS', compile_args)
 
         elem.add_item('LINK_ARGS', commands)
+        if mesonlib.is_os2() and isinstance(target, build.SharedLibrary):
+            # The library we will actually link to, which is an import library on OS/2 (not the DLL)
+            elem.add_item('IMPLIB', [self.get_target_filename_for_linking(target)])
```

Comment:
Is this significantly different enough from what Windows does that we can't re-use those paths for OS/2?

Reply:
I think we can. Where should I see ?

Reply:
Oooops, I've remembered what I forgot. On Windows import libraries are generated with compiler options. On the other hands, on OS/2 generation of import libraries require additional steps. So it's hard to re-use Windows code paths.

Reply:
Okay, then I'm okay leaving this as a separate path.